### PR TITLE
fix boot paramters update bug  GSoC

### PR DESCRIPTION
**Pull request by gqlo**
### What does this PR do?
This PR fixes a bug on updating existing boot parameters.  If users attempt to update an VM with existing boot parameters, it will fail. The reason is that  `found_tag = desc.find(tag)` will always return `None` since `tag` is not a direct children of `desc`. 

Moreover, `found_tag and found_tag.text != boot_tag_value`  returns an ElementTree  object which is not a Boolean value. As a result, if branch will evaluate to False and statements under it will never get executed .

### Tests written?
**[NOTICE] Bug fixes or features added to Salt require tests.**
Please review the [test documentation](https://docs.saltstack.com/en/latest/topics/tutorials/writing_tests.html) for details on how to implement tests into Salt's test suite.

Yes

### Commits signed with GPG?

Yes

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by cbosdo** [comment]
@MeAndTheFirefly Starting today you also need to run [black](https://github.com/psf/black) on your code to format it.

**Review by cbosdo — CHANGES_REQUESTED** [review]
The PR looks good to me, but there are two things needed:
 * Run black on your code
 * I would like to see the bug details you wrote in the PR description in the commit message too. Commit messages are easier to search and read than PRs when doing archeology.

**Comment by MeAndTheFirefly** [comment]
> The PR looks good to me, but there are two things needed:
> 
> * Run black on your code
> * I would like to see the bug details you wrote in the PR description in the commit message too. Commit messages are easier to search and read than PRs when doing archeology.

Roger that.  Thanks Cedric.

**Comment by cbosdo** [comment]
@MeAndTheFirefly Could you please squash the reformatting commit in the first one to make the review easier?

**Comment by MeAndTheFirefly** [comment]
> 

> @MeAndTheFirefly Could you please squash the reformatting commit in the first one to make the review easier?

sure, let me do that.

File: tests/unit/modules/test_virt.py [hunk 1]
```diff
@@ -1358,14 +1358,15 @@ def test_update(self):
...
-        self.assertEqual(
-            {
-                "definition": True,
-                "disk": {"attached": [], "detached": []},
-                "interface": {"attached": [], "detached": []},
-            },
-            virt.update("my vm", boot=boot),
-        )
+        self.assertEqual({
+                'definition': True,
```
Comment (cbosdo) [inline tests/unit/modules/test_virt.py:1362]: black should replace the `'` by `"` here.

**Review by cbosdo — CHANGES_REQUESTED** [review]
LGTM, but still some black-related problems in the test.

File: tests/unit/modules/test_virt.py (hunk 1 above)
Comment (waynew) [reply tests/unit/modules/test_virt.py:1362]: ☝️

**Comment by waynew** [comment]
@MeAndTheFirefly if you [install pre-commit](https://docs.saltstack.com/en/master/topics/development/contributing.html#quickstart) it will setup hooks that will automatically do the necessary formatting locally.

You might need to adapt the instructions a bit with pyenv:

```
$ pyenv install 3.7.7
$ eval "$(pyenv init -)"
$ pyenv shell 3.7.7
$ pre-commit install-hooks
```
If you've already got your commits applied, you'll want to also run:

```
$ pre-commit run --file <filename>
```

For the files in this PR 👍

File: tests/unit/modules/test_virt.py (hunk 1 above)
Comment (MeAndTheFirefly) [reply tests/unit/modules/test_virt.py]: Thanks waynew.  I just ran pre-commit and did a push.  :)

**Comment by MeAndTheFirefly** [comment]
> @MeAndTheFirefly if you [install pre-commit](https://docs.saltstack.com/en/master/topics/development/contributing.html#quickstart) it will setup hooks that will automatically do the necessary formatting locally.
> 
> You might need to adapt the instructions a bit with pyenv:
> 
> ```
> $ pyenv install 3.7.7
> $ eval "$(pyenv init -)"
> $ pyenv shell 3.7.7
> $ pre-commit install-hooks
> ```
> 
> If you've already got your commits applied, you'll want to also run:
> 
> ```
> $ pre-commit run --file <filename>
> ```
> 
> For the files in this PR

Thanks :) It works great!

**Review by cbosdo — APPROVED** [review]
Thanks for the fix!

**Comment by waynew** [comment]
Looks like some infrastructure failures on test build - rerunning now.

**Comment by MeAndTheFirefly** [comment]
> Looks like some infrastructure failures on test build - rerunning now.

Thanks waynew. It seems like we need to rerun another 2 builds.

**Pull request merged**